src/itemviewformaction.cpp

```diff
@@ -117,6 +118,19 @@ void ItemViewFormAction::prepare()
 					rxman,
 					"article");
 		} else {
+			// Add enclosure url to link list.
+			if (item->enclosure_type() != "") {
+				LinkType type;
+
+				if(item->enclosure_type().find("audio")) type = LinkType::AUDIO;
```

Comment:
Sorry I haven't foreseen this when I wrote pointers in #809. There are two problems with this approach:
1. it already misses some MIME types, e.g. "application/ogg" should be considered as audio type, but this code will classify it as a `LinkType::HREF`;
2. it's brittle: if we add more `LinkType` constants, or want to support more MIME types, we have to *remember* to update this code -- there is no compilation errors to guide us.

This code is intrinsically linked to [`utils::is_valid_podcast_type()`](https://github.com/newsboat/newsboat/blob/d72a9908ffda7d3d40921cf015ee69bb0a7dddd8/rust/libnewsboat/src/utils.rs#L415-L423) (via [Atom](https://github.com/newsboat/newsboat/blob/d72a9908ffda7d3d40921cf015ee69bb0a7dddd8/rss/atomparser.cpp#L125-L133) and [RSS](https://github.com/newsboat/newsboat/blob/d72a9908ffda7d3d40921cf015ee69bb0a7dddd8/rss/rss09xparser.cpp#L114-L141) parsers), and I think it makes sense to centralize the knowledge in that function. The way I see it is:
1. rename `enclosure_type()` to `enclosure_mime_type()`;
2. add new method, `enclosure_link_type()`, that returns `LinkType` (so now `RssItem` stores both a MIME string and a `LinkType`);
3. make `utils::is_valid_podcast_type()` return `Option<LinkType>` (on Rust side). On C++ side, this can be approximated with a signature like `bool is_valid_podcast_type(const std::string& mime_type, LinkType& link_type)`; it only fills out `link_type` if the function returns `true`;
4. replace the `if` ladder here with a call to `enclosure_link_type()`.

I remember you said you don't know C++, and now I'm asking you to write Rust :( It's fine if you leave this as-is, I'm sure someone will pick this up and finish it off quick enough.

Reply:
I'd be happy to help and I look always for opportunities to learn more about coding no matter what language it is. C++ or Rust is on my list for some time in the future. At the moment there is simply no time available for learning a new language. So I guess diving deeper in this would be a time consuming task for me and so I'd be glad if anybody else picks this up. Nevertheless maybe there is a way I can contribute in some other way?!

Reply:
Okay! I'll mention this on IRC, perhaps there's someone who's bored and will want to finish this. (I'm a bit busy with another PR.)

> Nevertheless maybe there is a way I can contribute in some other way?!

You could work on the docs. We have a couple issues written up that I think you could take up: #596, #475, #441.

Other than that, I'm not sure what else you could do. Do you have any ideas? Something you see about the project and think is wrong, something you think you could improve? Open an issue here or chat to me on IRC (Minoru in #newsboat on Freenode).